### PR TITLE
P4/tanner/move piece

### LINKED ISSUE #7: As soon as opponent joins game, it is the creator of the game's turn.
#76

### LINKED ISSUE #177: Move Game Piece

### PR DESCRIPTION
**Pull request by tking2096**
Closes #177 Closes #7

### CODE REVIEW

File: client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java [hunk 1]
```diff
@@ -44,44 +49,81 @@ public boolean canMovePieceAt(int row, int column) {
         return result;
     }
 
-    public boolean startGame() {
-        return true;
+    public void endGame() {
+
```
Comment (ALHennings) [inline client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java:53]: Why is there an empty method here?

File: client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/client/GamesManager.java [hunk 2]
```diff
@@ -33,6 +34,22 @@ public void setHomeController(HomeController homeController) {
...
+		if (game != null)
+			getAndShowGame(gameID, game.getViewingPlayer());
+	}
+	
+	public void handleGameEndedEvent(long gameID) {
+		JungleGame game = findById(gameID);
+		
+		if (game != null)
+			getAndShowGame(gameID, game.getViewingPlayer());
+		game = findById(gameID);
```
Comment (ALHennings) [inline client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/client/GamesManager.java:49]: Why do you run findById in both like 45 and 49. Won't it just set itself to the same value?

File: client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java [hunk 3]
```diff
@@ -44,44 +49,81 @@ public boolean canMovePieceAt(int row, int column) {
...
+        
+        GamesManager.getInstance().updateGame(this);
     }
 
-    public PlayerEnumType getViewingPlayer() {
-		return viewingPlayer;
-	}
+    public void quitGame() {
+        //This method should remove the user requesting it, if the game is not over that user officially loses the game.
+    }
```
Comment (ALHennings) [inline client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java:109]: Empty method?

**Review by ALHennings — APPROVED** [review]

File: client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java (hunk 3 above)
Comment (tking2096) [reply client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java:109]: Quit hasn't been implemented just yet.

File: client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java (hunk 1 above)
Comment (tking2096) [reply client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/game/JungleGame.java:53]: That will be filled when quit is implemented.

File: client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/client/GamesManager.java (hunk 2 above)
Comment (tking2096) [reply client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/client/GamesManager.java:49]: That first findById and game != null check would be better represented with localGameExists. The sencond findById gets the just-updated game.

**Pull request merged**